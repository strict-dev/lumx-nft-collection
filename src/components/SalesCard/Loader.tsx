import CardWrapper from 'components/CardWrapper'
import { GridLoader } from 'react-spinners'

export const ChartLoader = () => (
  <CardWrapper className="flex flex-col items-center justify-center h-64 gap-5 mb-2 bg-ne">
    <GridLoader color="#6e7191" />
  </CardWrapper>
)
