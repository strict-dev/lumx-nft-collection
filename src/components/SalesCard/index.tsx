import Container from 'components/Container'
import { Loader } from 'components/Loader'
import { useData } from 'hooks/useData'
import { useState } from 'react'
import { Chart } from 'react-chartjs-2'
import { chartOptions } from 'utils/chartUtils'

const SalesCard = () => {
  const [period, setPeriod] = useState<'24h' | '7d'>('24h')

  const { data, isLoading } = useData()

  return (
    <Container>
      <div className="flex flex-col gap-5 mb-2">
        <div className="flex flex-col gap-3">
          <div className="flex items-center justify-between">
            <h2 className="text-lg font-bold">Sales</h2>

            {isLoading ? (
              <div className="w-1/3 h-4 rounded-full bg-neutral-600 transit animate-pulse" />
            ) : (
              <p className="text-sm font-bold text-neutral-600">{`Showing ${data?.sales} sales`}</p>
            )}
          </div>
          <div className="flex items-center justify-between">
            <p className="text-xs font-bold text-neutral-700">Time:</p>
            <div>
              <button
                onClick={() => setPeriod('24h')}
                className={`px-4 py-2 font-bold rounded-sm 
                  transition duration-300 
                  ${period === '24h' && 'bg-neutral-600 text-white'}`}
              >
                24H
              </button>
              <button
                onClick={() => setPeriod('7d')}
                className={`px-4 py-2 font-bold rounded-sm text-neutral-600 transition duration-300 
                  ${period === '7d' && 'bg-neutral-600 text-white'}`}
              >
                7d
              </button>
            </div>
          </div>
        </div>
      </div>

      {isLoading ? (
        <Loader size="h-64" />
      ) : (
        <div className="flex flex-col gap-5 mb-2">
          <div className="w-full h-32">
            <Chart
              type="scatter"
              options={{
                scales: {
                  x: {
                    grid: {
                      display: false
                    },
                    display: false
                  },
                  y: {
                    grid: {
                      color: '#4E4B66'
                    },
                    ticks: {
                      font: {
                        size: 12,
                        weight: 'bold'
                      },
                      maxTicksLimit: 6
                    }
                  }
                },
                plugins: {
                  legend: {
                    display: false
                  }
                },
                maintainAspectRatio: false
              }}
              data={{
                datasets: [
                  {
                    data: [
                      ...(data?.scatterDataset1 as Array<{
                        x: number
                        y: number
                      }>)
                    ],
                    backgroundColor: 'rgba(255, 99, 132, 1)'
                  }
                ]
              }}
            />
          </div>

          <div className="flex items-center h-8">
            <Chart
              type="bar"
              options={{
                ...chartOptions,
                maintainAspectRatio: false
              }}
              data={{
                labels: [...(data?.barsDataset?.x as number[])],
                datasets: [
                  {
                    data: [...(data?.barsDataset.y as Array<number>)],
                    backgroundColor: '#D9DBE9'
                  }
                ]
              }}
            />
          </div>
        </div>
      )}
    </Container>
  )
}

export default SalesCard
