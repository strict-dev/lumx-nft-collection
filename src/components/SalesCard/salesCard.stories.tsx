import { Meta, Story } from '@storybook/react'

import SalesCard from '.'

export default {
  title: 'SalesCard',
  component: SalesCard
} as Meta

export const Default: Story = () => <SalesCard />
