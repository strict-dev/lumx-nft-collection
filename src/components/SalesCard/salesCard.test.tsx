import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render, screen } from '@testing-library/react'
import { RecoilRoot } from 'recoil'

import SalesCard from '.'

describe('<SalesCard />', () => {
  it('should render the heading', () => {
    const queryClient = new QueryClient()

    render(
      <QueryClientProvider client={queryClient}>
        <RecoilRoot>
          <SalesCard />
        </RecoilRoot>
      </QueryClientProvider>
    )

    expect(
      screen.getByRole('heading', {
        name: /sales/i
      })
    ).toBeInTheDocument()
  })
})
