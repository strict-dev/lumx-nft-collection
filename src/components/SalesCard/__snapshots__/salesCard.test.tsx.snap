// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SalesCard /> should render the heading 1`] = `
<div
  class="pb-60"
>
  <div
    class="mx-4 "
  >
    <div
      class="p-3 rounded-sm bg-neutral-800 flex flex-col gap-5 mb-2"
    >
      <div
        class="flex flex-col gap-3"
      >
        <div
          class="flex items-center justify-between"
        >
          <h2
            class="text-lg font-bold"
          >
            Sales
          </h2>
          <p
            class="text-sm font-bold text-neutral-600"
          >
            Showing undefined sales
          </p>
        </div>
        <div
          class="flex items-center justify-between"
        >
          <p
            class="text-xs font-bold text-neutral-700"
          >
            Time:
          </p>
          <div>
            <button
              class="px-4 py-2 font-bold rounded-sm 
                  transition duration-300 
                  bg-neutral-600 text-white"
            >
              24H
            </button>
            <button
              class="px-4 py-2 font-bold rounded-sm text-neutral-600 transition duration-300 
                  false"
            >
              7d
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mx-4 "
  >
    <div
      class="p-3 rounded-sm bg-neutral-800 flex flex-col gap-5 mb-2"
    >
      <canvas
        height="150"
        role="img"
        width="300"
      />
    </div>
  </div>
  <div
    class="mx-4 "
  >
    <div
      class="p-3 rounded-sm bg-neutral-800 flex flex-col gap-5 mb-2"
    >
      <canvas
        height="150"
        role="img"
        width="300"
      />
    </div>
  </div>
</div>
`;
